### ISSUE TITLE
MQTT "Connection lost" if I use a second device

### ISSUE THREAD

**Issue by Ersthelfer** [jomjol/AI-on-the-edge-device#296]
Hi, 
I've used a water meter with your project in the past and it worked fine. 
(You make a good job, thank you).

Today I put a gas meter into operation with your project.  Actually works too, but I now get an MQTT message from both meters about every 30 seconds with the topic /connection with the content "connection lost". 

A ping on both devices is answered correctly. There is also no reboot of the devices. 

Why do I get the "connection lost" message now? 

I have set an interval of 5 minutes. That's why I would have expected this message after 5 minutes at the earliest. 

If you want to contact me directly, use github@engel-city.de

Thanks a lot
Andreas

**Comment by jomjol** [comment]
I suppose you gave both of them different MQTT-Main Topics?

**Comment by Ersthelfer** [comment]
Yes, I use 
"Verbrauch/Frischwasser" and "Verbrauch/Gaszaehler". 

But even if I use "Frischwasser" and "Gaszaehler" the error remains :-(

**Comment by jomjol** [comment]
Vermute du sprichts Deutsch :-)
Wenn du die Version v8.0.x verwendest, dann solltest ud nur den maintopic setzen und der braucht keinen Subtyp. Dort reicht dann eigentlich "Verbrauch". Den Subnamen vergibst du über die "Numbers" in den ROI Einstellungen.

**Comment by Ersthelfer** [comment]
Du vermutest richtig ;-)

Wenn ich jetzt bei maintopic "Verbrauch" und als numbers "Gas" und "Wasser" benutze, ist der Fehler immer noch da.

Außerdem habe ich dann das gedankliche Problem, dass ich dann die Topics
- Verbrauch/connection
- Verbrauch/uptime
- Verbrauch/freeMem

nicht mehr den einzelnen ESP's zuordnen kann.

**Comment by Ersthelfer** [comment]
Ich habe noch etwas weiter geforscht:

Erster Aufbau 2 ESP am gleichen Netzteil -> Connection lost ca. alle 30 Sekunden. (egal welcher Main Topic)

Dritter ESP an einem eigenen Netzteil hinzu -> Connection lost ca. alle 30 Sekunden bei allen 3 Geräten

Die beiden ESP am gleichen Netzteil ausgeschaltet -> Das 3. Gerät verhält sich wie gewünscht / erwartet

Ein vierter ESP mit eigenem Netzteil hinzu -> Connection lost ca. alle 30 Sekunden bei Gerät 3 und 4 (1 und 2 war spannungslos)

Alle 4 Geräte eingeschaltet (2 am gleichen Netzteil + jeweils 2 mit eigenem Netzteil )-> Connection lost ca. alle 30 Sekunden bei allen 4 Geräten

D.H. sobald bei mir im Netzwerk mehr als ein Gerät online war gab es den Fehler....

**Comment by jomjol** [comment]
Vielleicht ist die Stromversorung nicht ausreichend? Wenn die Blitz-LED eingeschaltet ist, zieht der ESP bis zu 500mA.

**Comment by Ersthelfer** [comment]
Nein, die Stromversorgung ist es nicht. Die 2 einzelnen haben jeweils ein eigenes Netzteil 5V / 2A.
Auch würde dann der Ping abbrechen oder das Gerät rebooten, aber nichts in der Art passiert.

Alle 4 Geräte laufen einwandfrei. Es wird nur warum auch immer ein "connection lost" alle 30 Sekunden gesendet, sobald mehr als ein Gerät online ist.....

**Comment by Ersthelfer** [comment]
Helfen diese Fehlermeldungen evtl. weiter?

20:51:55.591 -> [0;31mE (56928) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128[0m
20:51:55.591 -> [0;31mE (56928) MQTT_CLIENT: mqtt_process_receive: mqtt_message_receive() returned -1[0m
20:51:55.591 -> [0;32mI (56938) interface_mqtt: MQTT_EVENT_DISCONNECTED[0m
20:52:05.610 -> [0;32mI (66948) interface_mqtt: MQTT_EVENT_BEFORE_CONNECT[0m
20:52:05.610 -> [0;32mI (66958) interface_mqtt: MQTT_EVENT_CONNECTED[0m
20:52:20.676 -> [0;31mE (82018) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128[0m
20:52:20.676 -> [0;31mE (82018) MQTT_CLIENT: mqtt_process_receive: mqtt_message_receive() returned -1[0m
20:52:20.676 -> [0;32mI (82028) interface_mqtt: MQTT_EVENT_DISCONNECTED[0m
20:52:35.681 -> [0;32mI (97028) interface_mqtt: MQTT_EVENT_BEFORE_CONNECT[0m
20:52:35.681 -> [0;32mI (97038) interface_mqtt: MQTT_EVENT_CONNECTED[0m
20:52:50.763 -> [0;31mE (112128) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128[0m
20:52:50.808 -> [0;31mE (112128) MQTT_CLIENT: mqtt_process_receive: mqtt_message_receive() returned -1[0m
20:52:50.808 -> [0;32mI (112138) interface_mqtt: MQTT_EVENT_DISCONNECTED[0m
20:53:05.788 -> [0;32mI (127138) interface_mqtt: MQTT_EVENT_BEFORE_CONNECT[0m
20:53:05.788 -> [0;32mI (127148) interface_mqtt: MQTT_EVENT_CONNECTED[0m
20:53:15.990 -> [0;31mE (137318) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128[0m
20:53:15.990 -> [0;31mE (137318) MQTT_CLIENT: mqtt_process_receive: mqtt_message_receive() returned -1[0m
20:53:15.990 -> [0;32mI (137328) interface_mqtt: MQTT_EVENT_DISCONNECTED[0m
20:53:30.982 -> [0;32mI (152328) interface_mqtt: MQTT_EVENT_BEFORE_CONNECT[0m
20:53:31.028 -> [0;32mI (152348) interface_mqtt: MQTT_EVENT_CONNECTED[0m
20:53:41.081 -> [0;31mE (162408) MQTT_CLIENT: mqtt_message_receive: transport_read() error: errno=128[0m
20:53:41.081 -> [0;31mE (162408) MQTT_CLIENT: mqtt_process_receive: mqtt_message_receive() returned -1[0m
20:53:41.081 -> [0;32mI (162418) interface_mqtt: MQTT_EVENT_DISCONNECTED[0m

**Comment by JanHBade** [comment]
ClientID = wasser

auf 2 unterschiedliche Werte gesetzt?

**Comment by jomjol** [comment]
ja - wäre jetzt auch meine Idee - was passiert dann?

**Comment by Ersthelfer** [comment]
Jepp, das war es. Super, danke....
Hit the user to continue ;-)

Kann geschlossen werden.

**Comment by Ersthelfer** [comment]
Da die ClientID normalerweise ja disabled ist, aber scheinbar trotzdem mit "wasser..." initialisiert wird, sollte man als default evtl. die Mac-Adresse eintragen, dann klappt es auch bei Dumpfbacken wie wir auf Anhieb ;-)

**Comment by amishv** [comment]
Please change the  clientname = "watermeter"; defined in ClassFlowMQTT.cpp in one of the devices to say watermeter1, else, MQTT will remove the previous client on new connection.

**Comment by jomjol** [comment]
@amishv : I don't get you point. You can name the MQTT clientname in the config.ini:

![grafik](https://user-images.githubusercontent.com/30766535/143721011-4a070694-9fda-46b7-b0ff-9ade297451b5.png)
